cda-comm-doip/src/ecu_connection.rs

```diff
@@ -57,12 +60,12 @@ pub(crate) struct EcuConnectionTarget {
 }
 
 impl ECUConnection for EcuConnectionVariant {
-    async fn send(&mut self, msg: DoipPayload) -> Result<(), String> {
+    async fn send(&mut self, msg: DoipPayload) -> Result<(), ConnectionError> {
         match self {
             EcuConnectionVariant::Tls(conn) => conn.send(msg).await,
             EcuConnectionVariant::Plain(conn) => conn.send(msg).await,
         }
-        .map_err(|e| format!("Failed to send message: {e:?}"))
+        .map_err(|e| ConnectionError::SendFailed(format!("Failed to send message: {e:?}")))
```

Comment:
suggestion: Using DiagServiceError here instead would prevent duplicating the SendFailed case